background.js

```diff
@@ -137,11 +137,84 @@ const MyQOnly = {
     return data.total_count;
   },
 
+  /**
+   * Is the current time within the user's working hours (if enabled)?
+   */
+  async isWorkingHours() {
+    console.log(`Checking working hours.`);
+
+    let { workingHours } = await browser.storage.local.get("workingHours");
+
+    if (typeof workingHours === 'undefined' || !workingHours.enabled) {
+      console.log(`Working hours are not enabled`);
+      return true;
+    }
+
+    let currentTime = new Date();
+
+    // It's possible for the start or end time to be an empty string, if the
+    // html5 time input had one empty field when a date checkbox was changed.
+    // The time input is kind of tricky to use; it's easy to overlook the
+    // am/pm chooser. Also, some people may just want to set days of the week,
+    // not times of day. In these cases, just skip the missing time check.
+    if (!workingHours.startTime) {
+      console.log(`Start time not set. Skipping start time check.`);
+    } else {
+      let startTime = new Date();
+      let [startHours, startMinutes] = workingHours.startTime.split(":");
+      startTime.setHours(startHours, startMinutes);
+      if (currentTime < startTime) {
+        console.log(`Current time (${currentTime.toLocaleTimeString()}) is earlier than start time (${startTime.toLocaleTimeString()})`);
+        return false;
+      }
+    }
+
+    if (!workingHours.endTime) {
+      console.log(`End time not set. Skipping end time check.`);
+    } else {
+      let endTime = new Date();
+      let [endHours, endMinutes] = workingHours.endTime.split(":");
+      endTime.setHours(endHours, endMinutes);
+      if (currentTime > endTime) {
+        console.log(`Current time (${currentTime.toLocaleTimeString()}) is later than end time (${endTime.toLocaleTimeString()})`);
+        return false;
+      }
+    }
+
+    // Unlike the times, workingHours.days should never be falsy: the days are
```

Comment:
Typo: "falsy" -> "false-y"

Reply:
Hmm, this is an interesting point. I've only ever seen `falsy`, and there is https://mdn.io/glossary/falsy, but OTOH, Google surprisingly turns up a ton of refs for `falsey`, and https://english.stackexchange.com/questions/109996/is-it-falsy-or-falsey definitely fails to resolve the ambiguity. But, hey, your project, your spelling :+1: